front/connectors/src/slack/slack.ts

```diff
@@ -0,0 +1,178 @@
+import { WebClient } from "@slack/web-api";
+import {
+  ConversationsListResponse,
+  Channel,
+} from "@slack/web-api/dist/response/ConversationsListResponse";
+import { ConversationsHistoryResponse } from "@slack/web-api/dist/response/ConversationsHistoryResponse";
+import axios, { AxiosRequestConfig } from "axios";
+
+export interface SlackConfig {
+  accessToken: string;
+}
+
+export interface DustConfig {
+  username: string;
+  datasourceId: string;
+  APIKey: string;
+}
+
+/**
+ * Fetches all channels for a Slack workspace and processes them.
+ * Calls to subsequent functions are made insde this function in order to handle
+ * pagination without blowing up the memory of the process.
+ * Another way to do that would be to use generator functions. TBD.
+ * @param slack_token Oauth access token or Bot token
+ * @param last_seen_ts only sync messages created after this timestamp.
+ */
+export async function processChannels(
+  slackConfig: SlackConfig,
+  dustConfig: DustConfig,
+  last_seen_ts?: string
```

Comment:
Is this set by Temporal?

Reply:
Right now you only process threads but thinking in advance in our ability to process main channel messages, I think we may need a bit more state than `last_seen_ts` or at least a bit more logic as we'll need to go back a bit in time to re-aggregate a full document for the channel messages according to whatever logic we decide for that?

What do you think?

Reply:
I think there's a lot to be designed here. Probably worth creating a section for each source in the design doc to lay out our strategy for each?

Reply:
I have started doing that. There is a section called "Synchronization design" and (only) one section for the Slack connector.